packages/lesswrong/server/authenticationMiddlewares.ts

```diff
@@ -395,12 +396,28 @@ export const addAuthMiddlewares = (addConnectHandler: AddMiddlewareType) => {
     })
   }
 
+  // This is a mock method used in playwright tests - it gets overwritten below
+  // for actual servers
+  let profileFromAccessToken: ProfileFromAccessToken = async (token: string) => {
+    const email = token.replace("access-token-", "");
+    return {
+      provider: "auth0",
+      id: email,
+      displayName: email,
+      emails: [{value: email}],
+      birthday: "",
+      _raw: email,
+      _json: {},
+    }
+  }
```

Comment:
So far I like this pattern of defining mock objects in the same file as the real object (in this case and with `MockAuth0Client`), as opposed to trying to patch and/or inject mocks from within the test. Maybe it will become unwieldy eventually but it saves a lot of complexity in mocking from the test side

Reply:
My hope is that we'll almost never need to mock anything (only external services) which is why I think this is OK. If we needed to mock a lot of stuff then, yeah, this'd probably get really messy really fast.